src/qibolab/instruments/rfsoc.py

```diff
@@ -436,17 +476,18 @@ def recursive_python_sweep(
         results = {}
         for idx in range(sweeper.expts):
             # update values
-            sweeper_parameter = sweeper.parameter[0].name.lower()
-            if sweeper_parameter in {
-                "amplitude",
-                "frequency",
-                "relative_phase",
-            }:
-                for jdx, _ in enumerate(sweeper.indexes):
-                    setattr(sequence[sweeper.indexes[jdx]], sweeper_parameter, values[jdx][idx])
-            else:
-                for kdx, jdx in enumerate(sweeper.indexes):
-                    qubits[jdx].flux.offset = values[kdx][idx]
+            for jdx, kdx in enumerate(sweeper.indexes):
+                sweeper_parameter = sweeper.parameter[jdx].name.lower()
+                if sweeper_parameter == "bias":
+                    qubits[kdx].flux.bias = values[jdx][idx]
+                elif sweeper_parameter in {
+                    "amplitude",
+                    "frequency",
+                    "relative_phase",
+                    "start",
+                    "duration",
+                }:
```

Comment:
Hi @AleCandido , thanks for the review! 
I don't think I understand this suggestion: what I would do maybe is having a function (in qibosoq):
```python
class Parameter(Enum):
	DURATION = ...
	...

	def variants(*parameters) -> set[Parameter]:
		return {getattr(self, name.upper()) for name in parameters}
```
Not sure if it would work like this, but more or less. Is this you are suggesting? What could I do with overload?

Reply:
> ```python
> class Parameter(Enum):
> 	DURATION = ...
> 	...
> 
> 	def variants(*parameters) -> set[Parameter]:
> 		return {getattr(self, name.upper()) for name in parameters}
> ```

Exactly this for `set`s (actually almost), but not only for them:
```py
from typing import overload

class Parameter(Enum):
    DURATION = ...
    ...

    @overload
    @classmethod
    def variants(cls, parameters: str) -> "Parameter":
        ...

    @overload
    @classmethod
    def variants(cls, parameters: set[str]) -> set["Parameter"]:
        ...

    @classmethod
    def variants(cls, parameters):
        if isinstance(parameters, str):
            return cls[parameters]
        if isinstance(parameters, set):
            return {cls[par] for par in parameters}
        ...
```

With `@overload` you can specify all the signatures, that is a subset of what you would specify with just `Union` (because `Union` in parameters and return type represents the cartesian product of options, while in this case you're telling "with parameter of type A returns A', while with parameter of type B returns B'").

Reply:
The other option, to make the spell out less options for the signature, is to use generics:

```py
from collections.abc import Iterable
from typing import TypeVar, overload

It = TypeVar("It", bound=Iterable)

class Parameter(Enum):
    DURATION = ...
    ...

    @overload
    @classmethod
    def variants(cls, parameters: str) -> "Parameter":
        ...

    @overload
    @classmethod
    def variants(cls, parameters: It[str]) -> It["Parameter"]:
        ...

    @classmethod
    def variants(cls, parameters):
        if isinstance(parameters, str):
            return cls[parameters]
        if isinstance(parameters, set):
            return {cls[par] for par in parameters}
        ...
```

Reply:
Cool, thanks! is it fair to do something like:
```python
@classmethod
    def variants(cls, parameters):
        if isinstance(parameters, str):
            return cls[parameters]
        if isinstance(parameters, Iterable):
            return type(parameters)(cls[par] for par in parameters)
```
so that is general for all the iterables?

In any case, I'll implement it in the open PR in qibosoq

Reply:
And also, shouldn't the type hint be 
```python
    def variants(cls, parameters: str) -> Parameter:
```

Reply:
I guess you might have figured out on your own, but the problem with using `Parameter` is that is not yet defined while you're defining the class (because it is the class, but it will only become available as a name after completing the definition, and this is inside - remember that Python is parsing line by line).

Because of this, the type `Self` has been introduced in py3.11... so we can not use it yet
https://docs.python.org/3/library/typing.html#typing.Self
That's why we use the string as a replacement (it's even mentioned in the above docs).